src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1310,7 +1311,7 @@ class CreateTableStatementSegment(ansi.CreateTableStatementSegment):
                 "COMMIT",
                 OneOf(Sequence("PRESERVE", "ROWS"), Sequence("DELETE", "ROWS"), "DROP"),
             ),
-            Sequence("TABLESPACE", Ref("TableReferenceSegment")),
+            Sequence("TABLESPACE", Ref("TablespaceReferenceSegment")),
```

Comment:
Are any linting rules affected by the changes like this? Do we need to look at updating other dialects to use the new types (for consistency)?

Reply:
I don't think any rules will be negatively affected by this. Some, in fact, will be positively affected as they're correctly identifying things as identifiers rather than parameters. The casing ones use identifier to identify case anyway. I think it's useful to be more specific though is we do want to start using the parser for interpretation. And deinitely yes is needs to be proliferated to all of the other dialects.

Reply:
Agree that I can only see positive rules impact here.

Also only ANSI has TABLESPACE and even then only a very light version and no test cases for that. Could be made more consistent but don't think massive priority given it's a very generic statement, lumped in with lots of others.